ote_cli/ote_cli/tools/explain.py

```diff
@@ -0,0 +1,168 @@
+        "--output",
+        default="saliency_dump",
+        help="Output path for explanation images.",
+    )
+    parser.add_argument(
+        "--load-weights",
+        required=True,
+        help="Load only weights from previously saved checkpoint",
+    )
+    parser.add_argument(
+        "--explain-model",
+        default="EigenCAM",
+        help=f"XAI model name, currently support {SUPPORTED_XAI_MODELS}",
+    )
+    add_hyper_parameters_sub_parser(parser, hyper_parameters, modes=("INFERENCE",))
+
+    return parser.parse_args(), template, hyper_parameters
+
+
+def get_explain_map(task, frame):
+    """
+    Returns list of explanations made by task on frame and time spent on doing explanation.
+    """
+
+    empty_annotation = AnnotationSceneEntity(
+        annotations=[], kind=AnnotationSceneKind.PREDICTION
+    )
+
+    item = DatasetItemEntity(
+        media=Image(cv2.cvtColor(frame, cv2.COLOR_BGR2RGB)),
+        annotation_scene=empty_annotation,
+    )
+
+    dataset = DatasetEntity(items=[item])
+
+    start_time = time.perf_counter()
+    explain_map = task.explain(
+        dataset,
+        InferenceParameters(is_evaluation=True),
+    )
+    elapsed_time = time.perf_counter() - start_time
+    saliency_map = explain_map[0]
+    return saliency_map, elapsed_time
+
+
+def main():
+    """
+    Main function that is used for model demonstration.
+    """
+
+    # Dynamically create an argument parser based on override parameters.
+    args, template, hyper_parameters = parse_args()
+    # Get new values from user's input.
+    updated_hyper_parameters = gen_params_dict_from_args(args)
+    # Override overridden parameters by user's values.
+    override_parameters(updated_hyper_parameters, hyper_parameters)
+
+    hyper_parameters = create(hyper_parameters)
+
+    # Get classes for Task, ConfigurableParameters and Dataset.
+    if args.load_weights.endswith(".pth"):
+        task_class = get_impl_class(template.entrypoints.base)
+    else:
+        raise ValueError(f"Unsupported file: {args.load_weights}.")
+
+    environment = TaskEnvironment(
+        model=None,
+        hyper_parameters=hyper_parameters,
+        label_schema=read_label_schema(args.load_weights),
+        model_template=template,
+    )
+
+    environment.model = read_model(
+        environment.get_model_configuration(), args.load_weights, None
+    )
+
+    task = task_class(task_environment=environment)
+
+    imgs = get_image_files(args.input)
+    if imgs is None:
+        raise ValueError(f"No image found in {args.input}")
+
+    if not os.path.exists(args.output):
+        os.makedirs(args.output)
+
+    elapsed_times = deque(maxlen=10)
+
+    for _, (root_dir, filename) in enumerate(imgs):
+        img = cv2.imread(os.path.join(root_dir, filename))
+        saliency_map, elapsed_time = get_explain_map(task, img)
+        elapsed_times.append(elapsed_time)
+        save_saliency_output(img, saliency_map.numpy, root_dir, filename.split[0])
```

Comment:
Please take a look into the saliency post-processing. The final result does not looks correct:
![image](https://user-images.githubusercontent.com/17028475/198058518-3831bb1b-0660-4f23-8fad-725e9088239b.png)

Reply:
Yes, fixed with rgb saliency map